subsys/bluetooth/host/att.c

```diff
@@ -176,35 +175,57 @@ static struct bt_att_req cancel;
  */
 static k_tid_t att_handle_rsp_thread;
 
-#define bt_att_tx_meta_data(buf) (((struct bt_att_tx_meta *)net_buf_user_data(buf))->data)
+static struct bt_att_tx_meta_data tx_meta_data_storage[CONFIG_BT_ATT_TX_COUNT];
 
-static struct bt_att_tx_meta_data tx_meta_data_storage[CONFIG_BT_CONN_TX_MAX];
-K_FIFO_DEFINE(free_att_tx_meta_data);
+struct bt_att_tx_meta_data *bt_att_get_tx_meta_data(const struct net_buf *buf);
+static void att_on_sent_cb(struct bt_att_tx_meta_data *meta);
 
-static struct bt_att_tx_meta_data *tx_meta_data_alloc(k_timeout_t timeout)
+static void att_tx_destroy(struct net_buf *buf)
 {
-	/* The meta data always get freed in the system workqueue,
-	 * so if we're in the same workqueue but there are no immediate
-	 * contexts available, there's no chance we'll get one by waiting.
+	struct bt_att_tx_meta_data *p_meta = bt_att_get_tx_meta_data(buf);
+	struct bt_att_tx_meta_data meta;
+
+	LOG_DBG("destroy %p", buf);
```

Comment:
I think the default behavior for `LOG_DBG` is to prefix the function, in which case the `destroy` here could be considered superfluous.